File: solutions/golang/coffeeVendingMachine/machine/coffeeFactory.go
package machine

import (
	"github.com/ashishps1/awesome-low-level-design/solutions/golang/coffeeVendingMachine/concreteCoffee"
	"github.com/ashishps1/awesome-low-level-design/solutions/golang/coffeeVendingMachine/enums"
	"github.com/ashishps1/awesome-low-level-design/solutions/golang/coffeeVendingMachine/interfaces"
)

type CoffeeFactory struct {
}

func (c *CoffeeFactory) GetCoffee(coffee enums.CoffeeType) interfaces.Coffee {
	switch coffee {
	case enums.CAPPUCCINO:
		return &concreteCoffee.Cappuccino{}
	case enums.MOCHA:
		return &concreteCoffee.Mocha{}
	case enums.LATTE:
		return &concreteCoffee.Latte{}
	}
	return nil
}

File: solutions/python/coffeevendingmachine/coffee_factory.py
from enums import CoffeeType
from coffee import Coffee, Espresso, Latte, Cappuccino

class CoffeeFactory:
    @staticmethod
    def create_coffee(coffee_type: CoffeeType) -> Coffee:
        if coffee_type == CoffeeType.ESPRESSO:
            return Espresso()
        elif coffee_type == CoffeeType.LATTE:
            return Latte()
        elif coffee_type == CoffeeType.CAPPUCCINO:
            return Cappuccino()
        else:
            raise ValueError(f"Unsupported coffee type: {coffee_type}")